tensorflow/core/kernels/mkl/mkl_avgpooling_op.cc

```diff
@@ -123,7 +123,14 @@ class MklAvgPoolingOp : public MklPoolingForwardOpBase<T> {
           dnnl::algorithm::pooling_avg_exclude_padding, pooling_prop_kind,
           static_cast<memory::format_tag>(this->data_format_mkldnn_), input_md,
           this->native_format_);
-      MklDnnThreadPool eigen_tp(context);
+      // Create the oneDNN wrapper over eigen threapool and set max threads
+      // in oneDNN.
+      Eigen::ThreadPoolInterface* eigen_interface =
+          context->device()
+              ->tensorflow_cpu_worker_threads()
+              ->workers->AsEigenThreadPool();
+      tsl::OneDnnThreadPool eigen_tp(eigen_interface,
+                                     ThreadPoolUseCallerThread());
```

Comment:
Can we define an `EigenThreadPoolFromTfContext` macro or inline function in `mkl_util.h` (or somewhere convenient)? Then we can create the threadpool in one statement.
```suggestion
      tsl::OneDnnThreadPool eigen_tp(EigenThreadPoolFromTfContext(context),
                                     ThreadPoolUseCallerThread());
```